mini-alloc/src/lib.rs

```diff
@@ -0,0 +1,91 @@
+#![no_std]
+
+use core::alloc::{GlobalAlloc, Layout};
+use core::cell::Cell;
+
+const PAGE_SIZE: usize = 0x10000;
+const PAGE_MASK: usize = !0xFFFF;
+
+fn heap_base() -> usize {
+    #[cfg(not(target_arch = "wasm32"))]
+    todo!();
+
+    #[cfg(target_arch = "wasm32")]
+    {
+        extern "C" {
+            // This symbol is created by the LLVM linker.
+            static __heap_base: u8;
+        }
+
+        unsafe { &__heap_base as *const u8 as usize }
+    }
+}
+
+#[derive(Default)]
+pub struct MiniAlloc {
+    start: Cell<usize>,
+}
+
+unsafe impl Sync for MiniAlloc {}
+
+impl MiniAlloc {
+    pub const INIT: Self = MiniAlloc {
+        start: Cell::new(0),
+    };
+}
+
+fn round_up_to_alignment(val: usize, align: usize) -> usize {
+    (val + align - 1) & (-(align as isize) as usize)
```

Comment:
`(val + align - 1) & !(align - 1)`

Also let's make it `const` :)

Reply:
`align` is always a power of two in `Layout`:

https://doc.rust-lang.org/std/alloc/struct.Layout.html

But I can add a comment on this function making it clear that `align` must be a power of two.